src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] dataFromFile = readFile(fromFileName);
+        String report = createReport(dataFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    private String[] readFile(String fromFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            int sumSupply = 0;
+            int sumBuy = 0;
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] split = line.split(",");
+                String operation = split[0];
+                int digit = Integer.parseInt(split[1]);
+
+                if (operation.equals(SUPPLY)) {
+                    sumSupply += digit;
+                } else {
+                    sumBuy += digit;
+                }
```

Comment:
You mixed file reading with some actions on the read data
Method with name readFile must only **read the file**